tinygrad/dtype.py

```diff
@@ -43,11 +43,14 @@ def cast_scalar(scalar: Scalar, dtype:DType):
 
 class dtypes:
   @staticmethod
-  def is_float(x: DType) -> bool: return x.scalar() in (dtypes.float16, dtypes.bfloat16, dtypes.float32, dtypes.float64)
-  @staticmethod # static methds on top, or bool in the type info will refer to dtypes.bool
-  def is_int(x: DType) -> bool: return x.scalar() in (dtypes.int8, dtypes.int16, dtypes.int32, dtypes.int64) or dtypes.is_unsigned(x)
+  @functools.lru_cache(None)
+  def is_float(x: DType) -> bool: return x.scalar() in set((dtypes.float16, dtypes.bfloat16, dtypes.float32, dtypes.float64))
```

Comment:
are you sure this `set()` is faster?

Reply:
It made the ops test much faster, but then lru_cache made it even faster. The set vs tuple is now negligible it appears. I can revert it.